test/src/androidTest/kotlin/io/realm/InstrumentedTests.kt

```diff
@@ -136,25 +149,156 @@ class InstrumentedTests {
+        assertTrue(mappedMemorySize < oneMB, "Opening a Realm should not cost more than 12KB")
+
+        // inserting ~ 100MB of data and keep a strong reference to all allocated objects
+        val referenceHolder = mutableListOf<Sample>()
+        realm!!.beginTransaction()
+        for (i in 1..100) {
+            realm.create(Sample::class).apply {
+                stringField = oneMBstring
+            }.also { referenceHolder.add(it) }
+        }
+        realm.commitTransaction()
+
+        mappedMemorySize = numberOfMemoryMappedBytes(command)
+        assertTrue(mappedMemorySize >= 99 * oneMB, "Committing the 100 objects should result in memory mapping ~ 99 MB. Current amount is ${bytesToHumanReadable(mappedMemorySize)}")
+
+        realm = null
+        triggerGC()
+
+        mappedMemorySize = numberOfMemoryMappedBytes(command)
+        assertTrue(mappedMemorySize >= 99 * oneMB, "Realm and its memory should still be allocated since we didn't release all the inserted objects yet. Current amount is ${bytesToHumanReadable(mappedMemorySize)}")
+
+        referenceHolder.clear()
+        triggerGC()
+
+        mappedMemorySize = numberOfMemoryMappedBytes(command)
+        assertTrue(mappedMemorySize < oneMB, "Releasing references should close the Realm and free all the 99 MB allocated previously. Current amount is ${bytesToHumanReadable(mappedMemorySize)}")
+    }
+
+    // make sure that calling realm.close() will force close the Realm and release native memory
+    @Test
+    fun closeShouldFreeMemory() {
+        val command = arrayListOf("/system/bin/sh", "-c", "cat /proc/${Process.myPid()}/maps | grep default.realm | awk '{print \$1}'")
+
+        val realm = openRealmFromTmpDir()
+
+        // Building a 1MB String
+        val oneMBstring = StringBuilder("").apply {
+            for (i in 1..4096) {
+                // 128 length (256 bytes)
+                append("v7TPOZtm50q8kMBoKiKRaD2JhXgjM6OUNzHojXuFXvxdtwtN9fCVIW4njdwVdZ9aChvXCtW4nzUYeYWbI6wuSspbyjvACtMtjQTtOoe12ZEPZPII6PAFTfbrQQxc3ymJ")
+            }
+        }.toString()
+        val oneMB = 1048576L
+
+        var mappedMemorySize = numberOfMemoryMappedBytes(command)
+        assertTrue(mappedMemorySize < oneMB, "Opening a Realm should not cost more than 12KB. Current amount is ${bytesToHumanReadable(mappedMemorySize)}")
+
+        // inserting ~ 100MB of data and keep a strong reference to all allocated objects
+        val referenceHolder = mutableListOf<Sample>()
+        realm.beginTransaction()
+        for (i in 1..100) {
+            realm.create(Sample::class).apply {
+                stringField = oneMBstring
+            }.also { referenceHolder.add(it) }
+        }
+        realm.commitTransaction()
+
+        mappedMemorySize = numberOfMemoryMappedBytes(command)
+        assertTrue(mappedMemorySize >= 99 * oneMB, "Committing the 100 objects should result in memory mapping of ~ 99 MB. Current amount is ${bytesToHumanReadable(mappedMemorySize)}")
```

Comment:
Should we have an upper bound, just to ensure that the test does what it is supposed to?